src/block/block_read.c

```diff
@@ -248,6 +249,7 @@ __wt_block_read_off(WT_SESSION_IMPL *session, WT_BLOCK *block,
 		bufsize = WT_MAX(size, buf->memsize + 10);
 	}
 	WT_RET(__wt_buf_init(session, buf, bufsize));
+	__wt_capacity_throttle(session, size, WT_THROTTLE_READ);
```

Comment:
> I'd be inclined to push the call to __wt_capacity_throttle into `__wt_read` and `__wt_write`- otherwise we aren't going to remember to track calls to read/write in the future.

This concerns me, because it implies that any locking in higher layers will potentially give us problems.

Reply:
I agree with Keith that throttling at the lowest level and holding a spinlock while we sleep may give us problems. Moving it to a higher level avoids that. Block manager and logging have been the only two paths out to the disk for many, many years. There exists the potential for another subsystem to be created that also does I/O sometime. But adding it down in `wt_read` or `wt_write` as insurance but impacting current, common code paths that might hold a lock seems like the worse tradeoff.

We could open a backlog ticket that did something like record the number of bytes coming through the throttle function (we already do that, but there are "nothing to do so return" paths that may skip it) and add in a comparison in `wt_read` or `wt_write` for total bytes written and triggers if those values are off by some amount as insurance.